### ISSUE TITLE
Dataset QA Workflow - PDX IMS Demographics

### ISSUE THREAD

**Issue by LMorgan-veraly** [hackoregon/neighborhoods-2018#12]
We need to move datasets through a QA workflow before uploading the (hopefully) finalized version to our AWS database.

QA tasks:

 

- [ ]  1. check for geocode errors (related to scope)
Use the below SQL syntax to filter the dataset to return only rows where the current geocoded location is not returning as inside our team’s scope (Portland metro): (replace <table_name> with dataset table name)

select
<tabel_name>.*
from <table_name>
where <table_name>.id not in
(select <table_name>.id
from <table_name>
join scope on ST_Contains (scope.geom, <table_name>.geom))

If the returned points are in fact not in Portland Metro, then everything is fine. However, if the point describes a location that is in the Portland metro, there is an error somewhere, probably in the geocoded location. Update the dataset to fix errors (document this process).

- [ ]  2. finalize dataset field names

Make sure spelling is correct and none are cut off.

Are the names logical? Could they be improved so that front end devs and anyone else in the future could more easily understand?

If changing field names: use lowercase, underscores, start all field names with a letter and no other symbols (such as + , . - and no spaces!).

- [ ]  3. address missing data

any columns missing data completely? Address issue or remove column from dataset if there is no data for that field

Check on null values, zeros, and blanks. Make sure decisions regarding missing data is documented and handled according to team processes.

- [ ]  4. Run other queries/procedures to aggregate and asses data to ensure dataset fidelity (especially regarding geocoding).

Once this is all done, mark issue as complete and send the updated dataset file to Zachary and Lynn. The database will then be updated.

**Comment by LMorgan-veraly** [comment]
IMS Demographic xlsx has #DIV/0 and check data comment on row 11 for ARLINGTON HEIGHTS/SYLVAN-HIGHLANDS. Seems like the issue is from 2000 nbr of households being 0. Zachary, how do you want to address this?

**Comment by LMorgan-veraly** [comment]
uploaded to AWS HO DB, needs to be added to Census API @accua . It is a csv file and neighborhood names correlate to the rlis_neighborhood boundary file. Once added to census API this issue can be closed.

**Comment by hassanshamim** [comment]
@accua here's the sql query for the join mentioned above.

```sql
SELECT ims.*, rlis.geom
	FROM  ims_nbrhd_demographics AS ims
	INNER JOIN rlis_neighborhoods as rlis
	ON ims.name = rlis.name
```

**Comment by hassanshamim** [comment]
addressed in #43
